src/components/SidebarNavigationItem/Collapsible.tsx

```diff
@@ -0,0 +1,167 @@
+  open?: boolean;
+  onOpenChange?: (value: boolean) => void;
+}
+
+type ContextProps = {
+  open: boolean;
+  onOpenChange: () => void;
+};
+
+const NavContext = createContext<ContextProps>({
+  open: false,
+  onOpenChange: () => null,
+});
+
+export const Collapsible = ({
+  open: openProp,
+  onOpenChange: onOpenChangeProp,
+  children,
+  ...props
+}: CollapsibleProps) => {
+  const [open, setOpen] = useState(openProp ?? false);
+  const onOpenChange = () => {
+    setOpen(open => {
+      if (typeof onOpenChangeProp === "function") {
+        onOpenChangeProp(!open);
+      }
+      return !open;
+    });
+  };
+
+  useEffect(() => {
+    setOpen(open => openProp ?? open);
+  }, [openProp]);
+
+  const value = {
+    open,
+    onOpenChange,
+  };
+  return (
+    <div {...props}>
+      <NavContext.Provider value={value}>{children}</NavContext.Provider>
+    </div>
+  );
+};
+
+const CollapsipleHeaderContainer = styled.div<{ $iconDir: IconDir }>`
+  margin-left: ${({ theme, $iconDir }) =>
+    $iconDir === "left" ? 0 : theme.click.image.sm.size.width};
+`;
+
+interface CollapsipleHeaderProps extends HTMLAttributes<HTMLDivElement> {
+  iconDir?: IconDir;
+}
+
+const CollapsipleHeader = ({ iconDir = "left", ...props }: CollapsipleHeaderProps) => {
+  return (
+    <CollapsipleHeaderContainer
+      $iconDir={iconDir}
+      {...props}
+    />
+  );
+};
+
+CollapsipleHeader.displayName = "CollapsibleHeader";
+Collapsible.Header = CollapsipleHeader;
+
+const CollapsipleTriggerButton = styled(EmptyButton)<{ $iconDir: IconDir }>`
+  display: flex;
+  align-items: center;
+  font: inherit;
+  color: inherit;
+  cursor: default;
+  [data-trigger-icon] {
+    visibility: hidden;
+    transition: transform 150ms cubic-bezier(0.87, 0, 0.13, 1);
+    &[data-open="true"] {
+      transform: rotate(90deg);
+    }
+  }
+  &:hover {
+    [data-trigger-icon] {
+      visibility: visible;
+    }
+  }
+`;
+export const FlexContainer = styled.span`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  gap: ${({ theme }) => theme.click.sidebar.navigation.item.default.space.gap};
+`;
+
+interface CollapsipleTriggerProps extends HTMLAttributes<HTMLButtonElement> {
+  iconDir?: IconDir;
+}
+
+const CollapsipleTrigger = ({
+  onClick: onClickProp,
+  children,
+  iconDir = "left",
+  ...props
+}: CollapsipleTriggerProps) => {
+  const { open, onOpenChange } = useContext(NavContext);
+  const onClick = (e: MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    e.stopPropagation();
+    if (onClickProp) {
+      onClickProp(e);
+    }
+    onOpenChange();
+  };
+
+  return (
+    <CollapsipleTriggerButton
+      onClick={onClick}
+      $iconDir={iconDir}
+      {...props}
+    >
+      {iconDir === "left" && (
+        <Icon
+          data-trigger-icon
+          name="chevron-right"
+          data-open={open.toString()}
+          size="sm"
+        />
+      )}
+      {children && <FlexContainer>{children}</FlexContainer>}
```

Comment:
what do you think of using a more semantically meaningful name?
maybe `LabelWrapper` or something similar

Reply:
This is a tempory wrapper until the flex and grid options are added
After that this would be deleted
I'll rename it to LabelWrapper for now
I named it like Flex inorder to search and replace it faster